tests/BenchmarkDotNet.Tests/Mocks/MockHelper.cs

```diff
@@ -0,0 +1,27 @@
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Exporters;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using Xunit.Abstractions;
+
+namespace BenchmarkDotNet.Tests.Mocks
+{
+    public static class MockHelper
+    {
+        public static Summary Run<T>(ITestOutputHelper output)
```

Comment:
We could make this method accept a mandatory `Func<BenchmarkCase, List<Measurement>>` argument, pass it to `MockToolchain` instance and remove the need of having a custom attribute, interface and reflection.

Reply:
@adamsitnik good idea, I like it! I rewrote the implementation, could you please check out the new version?